GsaGH/Helpers/GH/RhinoConversions.cs

```diff
@@ -61,12 +62,27 @@ public class RhinoConversions {
 
       tolerance *= 2;
       brep = Brep.CreatePlanarBreps(curves, tolerance);
-      if (brep != null) {
+      if (brep != null && brep.Length > 0 && brep[0].IsValid) {
         return brep[0];
       }
 
-      var brep2 = Brep.CreateEdgeSurface(curves);
-      return brep2;
+      if (curves.Count < 5) {
+        var brep2 = Brep.CreateEdgeSurface(curves);
+        if (brep2 != null && brep2.IsValid) {
+          return brep2;
+        }
+      }
+
+      var stopwatch = Stopwatch.StartNew();
+      while (brep == null || brep.Length == 0 || !brep[0].IsValid) {
+        if (stopwatch.ElapsedMilliseconds > 5000) { break; }
```

Comment:
is that one of those magic numbers you mentioned lately? ;)
also please add a line break after an opening brace..